contracts/src/v0.8/EIP_712_1014_4337/interfaces/UserOperation.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+/* solhint-disable no-inline-assembly */
+
+    /**
+     * User Operation struct
+     * @param sender the sender account of this request.
+     * @param nonce unique value the sender uses to verify it is not a replay.
+     * @param initCode if set, the account contract will be created by this constructor/
+     * @param callData the method call to execute on this account.
+     * @param callGasLimit the gas limit passed to the callData method call.
+     * @param verificationGasLimit gas used for validateUserOp and validatePaymasterUserOp.
+     * @param preVerificationGas gas not calculated by the handleOps method, but added to the gas paid. Covers batch overhead.
+     * @param maxFeePerGas same as EIP-1559 gas parameter.
+     * @param maxPriorityFeePerGas same as EIP-1559 gas parameter.
+     * @param paymasterAndData if set, this field holds the paymaster address and paymaster-specific data. the paymaster will pay for the transaction instead of the sender.
+     * @param signature sender-verified signature over the entire request, the EntryPoint address and the chain ID.
+     */
+    struct UserOperation {
```

Comment:
This is interesting - didn't know you could do struct declarations outside of `contract` or `library` scopes 💭